### ISSUE TITLE
typescript inteli-sense constant crashes after update to 1.1.13 on Mac Sierra.

### ISSUE THREAD

**Issue by pitAlex** [Microsoft/vscode#28824]
Hello,
After upgrading vscode to the latest version the typescript inteli-sense crashes constantly providing me with no autocomplete or info as I work. This is the output:

```
Using tsserver from: /usr/local/lib/node_modules/typescript/lib/tsserver.js
[Info  - 5:34:06 PM] Forking TSServer
PATH: /Users/****/Library/Android/sdk/tools:/Users/****/Library/Android/sdk/platform-tools:/usr/local/Cellar/php56/5.6.20/bin:/usr/local/mysql-5.5.48-osx10.8-x86_64/bin:/usr/local/Cellar/php56/5.6.20/bin/pear:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands
[Error  - 5:41:46 PM] 'quickinfo' request failed with error.
Error: Cancelled Request 25
    at TypeScriptServiceClient.tryCancelRequest (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript/out/typescriptServiceClient.js:755:25)
    at token.onCancellationRequested (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript/out/typescriptServiceClient.js:695:30)
    at e.invoke (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:14715)
    at e.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:53565)
    at e.cancel (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:56383)
    at e.cancel (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:56993)
    at n.Class.derive._oncancel.<anonymous> (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:57458)
    at n.Class.derive._oncancel._cancelAction (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:46586)
    at Object.enter (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:42343)
    at n.Class.derive._oncancel._run (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:44654)

[Error  - 7:20:23 PM] 'completionEntryDetails' request failed with error.
Error: Cancelled Request 38
    at TypeScriptServiceClient.tryCancelRequest (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript/out/typescriptServiceClient.js:755:25)
    at token.onCancellationRequested (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript/out/typescriptServiceClient.js:695:30)
    at e.invoke (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:14715)
    at e.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:53565)
    at e.cancel (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:56383)
    at e.cancel (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:56993)
    at n.Class.derive._oncancel.<anonymous> (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:57458)
    at n.Class.derive._oncancel._cancelAction (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:46586)
    at Object.enter (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:42343)
    at n.Class.derive._oncancel._run (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:44654)
```

**Comment by steelsojka** [comment]
I am seeing this as well though with a slightly different error. This happens even after disabling all extensions.
```
[Error  - 2:52:39 PM] 'signatureHelp' request failed with error.
Error processing request. Cannot read property 'charCodeAt' of undefined
TypeError: Cannot read property 'charCodeAt' of undefined
    at getRootLength (/Users/****/Projects/develop/node_modules/typescript/lib/tsserver.js:2237:17)
    at Object.isRootedDiskPath (/Users/****/Projects/develop/node_modules/typescript/lib/tsserver.js:2347:16)
    at moduleHasNonRelativeName (/Users/****/Projects/develop/node_modules/typescript/lib/tsserver.js:4335:21)
    at tryLoadModuleUsingOptionalResolutionSettings (/Users/****/Projects/develop/node_modules/typescript/lib/tsserver.js:4637:13)
    at tryResolve (/Users/****/Projects/develop/node_modules/typescript/lib/tsserver.js:4766:28)
    at nodeModuleNameResolverWorker (/Users/****/Projects/develop/node_modules/typescript/lib/tsserver.js:4758:68)
    at nodeModuleNameResolver (/Users/****/Projects/develop/node_modules/typescript/lib/tsserver.js:4749:16)
    at Object.resolveModuleName (/Users/****/Projects/develop/node_modules/typescript/lib/tsserver.js:4611:30)
    at LSHost.resolveModuleName (/Users/****/Projects/develop/node_modules/typescript/lib/tsserver.js:73502:44)
    at LSHost.resolveNamesWithLocalCache (/Users/****/Projects/develop/node_modules/typescript/lib/tsserver.js:73539:42)
[Error  - 2:52:41 PM] 'completions' request failed with error.
Error: Cancelled Request 20
    at TypeScriptServiceClient.tryCancelRequest (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript/out/typescriptServiceClient.js:755:25)
    at token.onCancellationRequested (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript/out/typescriptServiceClient.js:695:30)
    at e.invoke (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:14715)
    at e.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:53565)
    at e.cancel (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:56383)
    at e.cancel (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:56993)
    at n.Class.derive._oncancel.<anonymous> (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:57458)
    at n.Class.derive._oncancel._cancelAction (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:46586)
    at Object.enter (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:42343)
    at n.Class.derive._oncancel._run (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:44654)
```
Platform: Mac 10.10.4

**Comment by pitAlex** [comment]
I get @steelsojka's error as well. The ones I provided are just the most frequent ones. In the 'charCodeAt' case, after it triggers and I hover over some variables it gives information about variables on the next line or that are 3 or 4 positions behind my current pointer. It usually triggers after I done some keyboard combination that have "CMD" or "ALT" key included.

**Comment by mjbvz** [comment]
The `charCodeAt` error is tracked upstream by https://github.com/Microsoft/TypeScript/issues/16048 If you have code or steps that can reliably reproduce the error, please add it to that issue 

The cancellation errors are expected. in 1.13 we incorrectly logged them in a way that looks like errors. This should be fixed in insiders #28095

**Comment by pitAlex** [comment]
Aha,

@mjbvz why don't you take a look at my attached demo and see how wierd the editor is acting after one of the errors are trigger. Notice how near the end I undo everything then re-open the file and thats when things get back to normal, temporarily. 
[demo.mpg.zip](https://github.com/Microsoft/vscode/files/1082084/demo.mpg.zip)
